hazelcast/src/main/java/com/hazelcast/internal/usercodedeployment/impl/ClassData.java

```diff
@@ -19,31 +19,48 @@
 import com.hazelcast.nio.ObjectDataInput;
 import com.hazelcast.nio.ObjectDataOutput;
 import com.hazelcast.nio.serialization.IdentifiedDataSerializable;
-import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+import com.hazelcast.nio.serialization.impl.Versioned;
+import com.hazelcast.version.Version;
 
 import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+import static com.hazelcast.internal.cluster.Versions.V3_9;
 
 /**
- * Bytecode of a class.
+ * Carries byte code of a class along with its inner classes.
+ *
  *
  * It's wrapped inside own object as it allows to add additional metadata and maintain compatibility
  * with Hazelcast Rolling Upgrade.
  */
-public class ClassData implements IdentifiedDataSerializable {
+public class ClassData implements IdentifiedDataSerializable, Versioned {
+
+    private Map<String, byte[]> classDefinitions;
 
-    private byte[] classDefinition;
+    //this field needed for compatibility with 3.8
+    private byte[] requestedClassDefinition;
```

Comment:
I think you can keep this field even after 3.9. Reason is, mostly there'll be only a single class definition. Inner class definitions will be mostly empty. So, if you rename these fields something like `mainClassDefinition` and `innerClassDefinitions`...